### PR TITLE
Add deb packaging directives, a README.txt, Copyright.txt, and a currently unused CMake module for GFlags.

### PR DESCRIPTION
**Pull request by nashef**
Specific changes included:

- Adds GFLAGS as a dependency in the top-level CMakeLists.txt
- Adds a `Copyright.txt` with the original BSD-style copyright content.
- Adds a README.txt.
- Adds `FindGFLAGS.cmake` to the modules directory.
- Return to building a library and linking `main.cc` to it.

### CODE REVIEW

File: rosette/CMakeLists.txt [hunk 1]
```diff
@@ -31,10 +31,47 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-invalid-offsetof")
...
 include_directories("${PROJECT_SOURCE_DIR}/h")
 include_directories("${PROJECT_SOURCE_DIR}/h/sys")
 add_subdirectory (src)
 add_subdirectory (rbl/rosette)
 
 include_directories("${PROJECT_BINARY_DIR}")
 
+INCLUDE(InstallRequiredSystemLibraries)
+
+SET(CPACK_PACKAGE_DESCRIPTION_SUMMARY "The Pyr8 Server Environment")
```
Comment (kirkwood) [inline rosette/CMakeLists.txt]: Is this copy pasta?

**Comment by kirkwood** [comment]
```
<gauss:...ain/rosette (remotes/nash/dev)> ./clean.sh && ./build.sh 
+ rm -rf build.out
+ rm -rf build.third-party
+ '[' '!' -h third-party ']'
+ rm -rf third-party
+ THIRD_PARTY_BUILD=build.third-party
+++ pwd
++ readlink -f /tmp/rchain/rosette
+ DEF_PFX=/tmp/rchain/rosette/build.out
+ BUILD=/tmp/rchain/rosette/build.out
+ LIBDIR=build.third-party/lib
+ export VERBOSE=1
+ VERBOSE=1
+ export DEBUG=1
+ DEBUG=1
+ CMAKE_VARS=-DCMAKE_BUILD_TYPE=/tmp/rchain/rosette/build.out
+ PROCS_ARG='-j 4'
+++ pwd
++ readlink -f /tmp/rchain/rosette
+ SRCDIR=/tmp/rchain/rosette
+ '[' '!' -d /tmp/rchain/rosette/build.out ']'
+ mkdir -p /tmp/rchain/rosette/build.out
+ cd /tmp/rchain/rosette/build.out
+ cmake -B/tmp/rchain/rosette/build.out -H/tmp/rchain/rosette
Re-run cmake no build system arguments
-- The CXX compiler identification is GNU 7.2.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test COMPILER_SUPPORTS_CXX11
-- Performing Test COMPILER_SUPPORTS_CXX11 - Success
-- Performing Test COMPILER_SUPPORTS_CXX0X
-- Performing Test COMPILER_SUPPORTS_CXX0X - Success
CMake Error at /usr/share/cmake-3.9/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find GFLAGS (missing: GFLAGS_INCLUDE_DIR GFLAGS_LIBRARY)
Call Stack (most recent call first):
  /usr/share/cmake-3.9/Modules/FindPackageHandleStandardArgs.cmake:377 (_FPHSA_FAILURE_MESSAGE)
  cmake/Modules/FindGFLAGS.cmake:41 (find_package_handle_standard_args)
  CMakeLists.txt:34 (find_package)


-- Configuring incomplete, errors occurred!
See also "/tmp/rchain/rosette/build.out/CMakeFiles/CMakeOutput.log".
```

File: rosette/CMakeLists.txt (hunk 1 above)
Comment (nashef) [reply rosette/CMakeLists.txt]: Oops, yes. Thanks for catching that.

**Comment by nashef** [comment]
@kirkwood Yes, try `apt-get install libgflags` to resolve your new dependency problem.

**Comment by nashef** [comment]
Maybe also requires `libgflags-dev` on Ubuntu, unsure. Once you figure out what works, please let me know so that I can look at adding it to the `README.txt` and `CMakeLists.txt` files.

**Comment by kjmcqueen** [comment]
root@ubuntu16VM:/media/sf_kmcqueen/proj/nch/rosette# apt-get install libgflags
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package libgflags

**Comment by kjmcqueen** [comment]
Installing libgflags-dev helped.

Now I get:

    [100%] Linking CXX executable rosette
cd /media/sf_kmcqueen/proj/nch/rosette/build.out/src && /usr/bin/cmake -E cmake_link_script CMakeFiles/rosette.dir/link.txt --verbose=1
/usr/bin/c++    -std=c++11 -DGCC27X -ggdb -m32 -Wall -Werror -Wno-write-strings -Wno-invalid-offsetof -Wno-delete-incomplete -Wno-deprecated-declarations   CMakeFiles/rosette.dir/main.cc.o  -o rosette librosettecore.a /usr/lib/x86_64-linux-gnu/libgflags.so 
/usr/lib/x86_64-linux-gnu/libgflags.so: error adding symbols: File in wrong format
collect2: error: ld returned 1 exit status
src/CMakeFiles/rosette.dir/build.make:96: recipe for target 'src/rosette' failed
make[2]: *** [src/rosette] Error 1
make[2]: Leaving directory '/media/sf_kmcqueen/proj/nch/rosette/build.out'
CMakeFiles/Makefile2:124: recipe for target 'src/CMakeFiles/rosette.dir/all' failed
make[1]: *** [src/CMakeFiles/rosette.dir/all] Error 2
make[1]: Leaving directory '/media/sf_kmcqueen/proj/nch/rosette/build.out'
Makefile:116: recipe for target 'all' failed
make: *** [all] Error 2

File: rosette/cmake/Modules/FindGFLAGS.cmake [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# - Try to find GFLAGS
+#
+# The following variables are optionally searched for defaults
+#  GFLAGS_ROOT_DIR:            Base directory where all GFLAGS components are found
+#
+# The following are set after configuration is done: 
+#  GFLAGS_FOUND
+#  GFLAGS_INCLUDE_DIRS
+#  GFLAGS_LIBRARIES
+#  GFLAGS_LIBRARYRARY_DIRS
```
Comment (kirkwood) [inline rosette/cmake/Modules/FindGFLAGS.cmake]: Typoypo?

Comment (nashef) [reply rosette/cmake/Modules/FindGFLAGS.cmake]: Fixed.

**Comment by nashef** [comment]
https://packages.ubuntu.com/trusty/libgflags2

**Comment by kirkwood** [comment]
After installing libgflags-dev:i386, I need to make this change to get it to compile.
```diff
1 file changed, 3 insertions(+)
rosette/CMakeLists.txt | 3 +++

modified   rosette/CMakeLists.txt
@@ -31,6 +31,9 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-invalid-offsetof")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-delete-incomplete")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-declarations")
 
+SET(CMAKE_LIBRARY_PATH "/usr/lib/i386-linux-gnu")
+include_directories(BEFORE /usr/include/i386-linux-gnu)
+
 find_package(GFLAGS REQUIRED)
 include_directories(${GFLAGS_INCLUDE_DIRS})
 set(LIBS ${LIBS} ${GFLAGS_LIBRARIES})
```
otherwise it fails to find the required library with
```
CMake Error at /usr/share/cmake-3.9/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find GFLAGS (missing: GFLAGS_LIBRARY)
Call Stack (most recent call first):
  /usr/share/cmake-3.9/Modules/FindPackageHandleStandardArgs.cmake:377 (_FPHSA_FAILURE_MESSAGE)
  cmake/Modules/FindGFLAGS.cmake:41 (find_package_handle_standard_args)
  CMakeLists.txt:34 (find_package)
```

**Comment by kirkwood** [comment]
Here is a different option (not involving specific paths) which seems to cause cmake to find 32-bit libraries:
```diff
diff --git a/rosette/build.sh b/rosette/build.sh
index 6f5dcf9..aed1f81 100755
--- a/rosette/build.sh
+++ b/rosette/build.sh
@@ -14,7 +14,7 @@ BUILD="${2:-${DEF_PFX}}"
 LIBDIR="${THIRD_PARTY_BUILD}/lib"
 export VERBOSE=1
 export DEBUG=1
-CMAKE_VARS="-DCMAKE_BUILD_TYPE=${BUILD:-Release}"
+CMAKE_VARS="-DCMAKE_BUILD_TYPE=${BUILD:-Release} -DCMAKE_CXX_FLAGS=-m32 -DCMAKE_SHARED_LINKER_FLAGS=-m32"
 PROCS_ARG="-j ${BUILD_PROCS:-4}"
 
 SRCDIR=$(readlink -f $(pwd))
@@ -23,7 +23,7 @@ if [ ! -d ${BUILD} ] ; then {
 } ; fi
 
 cd ${BUILD}
-cmake -B${BUILD} -H${SRCDIR}
+cmake ${CMAKE_VARS} -B${BUILD} -H${SRCDIR}
 make ${PROCS_ARG}
 make test
 make package
```

**Comment by kjmcqueen** [comment]
Works for me.

**Pull request merged**